From 316bf38fc8d17824bfd67c732b54152ac840cf1e Mon Sep 17 00:00:00 2001
From: Giuseppe Congiu <gcongiu@anl.gov>
Date: Thu, 25 Jul 2019 21:47:00 -0500
Subject: [PATCH] pm/hydra: include alpha-only nodelist strings

Current slurm nodelist parsing in hydra only considers cases in which
the node basename is alpha-numeric, i.e., host00. Cases in which the
nodelist contains alpha-only strings, i.e., host are not caught. This
patch fixes this behaviour.
---
 src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c b/src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c
index 47d07a162d..817fd8ae64 100644
--- a/src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c
+++ b/src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c
@@ -92,8 +92,8 @@ static HYD_status list_to_nodes(char *str)
     /* compile element regex for old format: "h14" */
     regcomp(&ematch_old, "([a-zA-Z]+[0-9]+)", REG_EXTENDED);
 
-    /* compile group-0 regex for new format: "h00-[00-12,14] | h00[00-12,14] | h00-14 | h0014" */
-    regcomp(&gmatch_new[0], "(,|^)([a-zA-Z0-9]+-?)(\\[[-,0-9]+\\]|[0-9]*)(,|$)", REG_EXTENDED);
+    /* compile group-0 regex for new format: "h00-[00-12,14] | h00[00-12,14] | h00-14 | h0014 | h-[00-12,14] | h[00-12,14] | h-14 | h" */
+    regcomp(&gmatch_new[0], "(,|^)([a-zA-Z]+[0-9]*-?)(\\[[-,0-9]+\\]|[0-9]*)(,|$)", REG_EXTENDED);
 
     /* compile group-1 regex for new format: "00-12 | 14 | " */
     regcomp(&gmatch_new[1], "([[,]|^)([0-9]+-[0-9]+|[0-9]*)([],]|$)", REG_EXTENDED);
